apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -1672,3 +1687,55 @@ type APIManagerList struct {
 func init() {
 	SchemeBuilder.Register(&APIManager{}, &APIManagerList{})
 }
+
+func (apimanager *APIManager) IsSystemRedisTLSEnabled() bool {
+	tlsEnabled := false
+	if apimanager.Spec.SystemRedisTLSEnabled != nil &&
+		*apimanager.Spec.SystemRedisTLSEnabled &&
+		apimanager.Spec.ExternalComponents != nil &&
+		apimanager.Spec.ExternalComponents.System != nil &&
+		*apimanager.Spec.ExternalComponents.System.Redis &&
+		apimanager.Spec.ExternalComponents.Backend != nil &&
+		*apimanager.Spec.ExternalComponents.Backend.Redis {
+		tlsEnabled = true
+	}
+	return tlsEnabled
+}
+func (apimanager *APIManager) IsBackendRedisTLSEnabled() bool {
+	tlsEnabled := false
+	if apimanager.Spec.BackendRedisTLSEnabled != nil &&
+		*apimanager.Spec.BackendRedisTLSEnabled &&
+		apimanager.Spec.ExternalComponents != nil &&
+		apimanager.Spec.ExternalComponents.Backend != nil &&
+		*apimanager.Spec.ExternalComponents.Backend.Redis {
+		tlsEnabled = true
+	}
+	return tlsEnabled
+}
+func (apimanager *APIManager) IsQueuesRedisTLSEnabled() bool {
+	tlsEnabled := false
+	if apimanager.Spec.QueuesRedisTLSEnabled != nil &&
+		*apimanager.Spec.QueuesRedisTLSEnabled &&
+		apimanager.Spec.ExternalComponents != nil &&
```

Comment:
external components check should not be here. All installations of 3scale on 2.16 + must use external.

Reply:
Done. Thanks